server/src/main/java/org/apache/druid/server/coordinator/duty/KillUnusedSegments.java

```diff
@@ -313,4 +327,40 @@ private int getAvailableKillTaskSlots(final CoordinatorDynamicConfig config, fin
     stats.add(Stats.Kill.MAX_SLOTS, killTaskCapacity);
     return availableKillTaskSlots;
   }
+
+  /**
+   * Return "intervals" limited to only the intervals that are contained within "maxPeriod" of the earliest start
+   * point of all "intervals". One exception: the very earliest-starting intervals are always included, regardless of
+   * whether they are contained in "maxPeriod".
+   *
+   * @param intervals intervals list
+   * @param maxPeriod period for limiting intervals. If zero, returns all intervals.
+   */
+  static List<Interval> limitToPeriod(final List<Interval> intervals, final Period maxPeriod)
+  {
+    if (DateTimes.EPOCH.plus(maxPeriod).equals(DateTimes.EPOCH)) {
+      // maxPeriod is zero, return all intervals.
+      return intervals;
+    } else {
```

Comment:
short-circuit if size is 0 or 1.

```suggestion
    } else if (intervals.size() <= 1) {
        return intervals;
    } else {
```